crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -249,24 +249,42 @@ impl<P: SystemParam + 'static> System for ParamSystem<P> {
 ///
 /// fn my_system_function(an_usize_resource: Res<usize>) {}
 ///
-/// let system = my_system_function.system();
+/// let system = IntoSystem::system(my_system_function);
 /// ```
 // This trait has to be generic because we have potentially overlapping impls, in particular
 // because Rust thinks a type could impl multiple different `FnMut` combinations
 // even though none can currently
-pub trait IntoSystem<In, Out, Params> {
+pub trait IntoSystem<In, Out, Params>: Sized {
     type System: System<In = In, Out = Out>;
     /// Turns this value into its corresponding [`System`].
-    fn system(self) -> Self::System;
+    ///
+    /// Use of this method was formerly required whenever adding a `system` to an `App`.
+    /// or other cases where a system is required.
+    /// However, since [#2398](https://github.com/bevyengine/bevy/pull/2398),
+    /// this is no longer required.
+    ///
+    /// In future, this method will be removed.
+    ///
+    /// One valid use of this method is to assert that a given function is a valid system.
+    /// For this case, use `IntoSystem::into_system` instead.
+    #[deprecated(
+        since = "0.6.0",
+        note = "In most cases, using `.system()` is no longer necessary, and so it can be removed"
```

Comment:
This reads as if there are still cases that require `.system()`. Is that the case? If so, what happens with them when we remove it (like announced in the docs)? Should we actually deprecate it then?

Reply:
This is the case explained directly in the line above, which is testing that a function is a system.